ziti/pki/store/local.go

```diff
@@ -217,18 +217,23 @@ func (l *Local) writeBundle(caName, name string, isCa bool, key, cert []byte) er
 }
 
 // writeChainBundle concats...
-func (l *Local) writeChainBundle(caName, name string, chainName string) error {
+func (l *Local) writeChainBundle(caName, destCaName, name string, chainName string) error {
 	caDir := filepath.Join(l.Root, caName)
 	if _, err := os.Stat(caDir); err != nil {
 		if err := InitCADir(caDir); err != nil {
 			return fmt.Errorf("root directory for CA %v does not exist and cannot be created: %v", caDir, err)
 		}
 	}
 
-	caPath := filepath.Join(l.Root, caName, LocalCertsDir, caName+".cert")
-	caIn, err := os.Open(caPath)
+	caChainPath := filepath.Join(l.Root, caName, LocalCertsDir, caName+".chain.pem")
+	caIn, err := os.Open(caChainPath)
 	if err != nil {
-		return fmt.Errorf("failed to open CA: %v: %v", caPath, err)
+		// no chain path, likely means we're using the root as the CA, so fallback to the cert
```

Comment:
this feels like a fail/panic situation to me? i kinda feel like this should be 'right' or not and fail without making inferences?

Reply:
The cases are relatively well defined. If the root is the signer, there will be no chain, if the signer is an intermediate there will be. Either way, if there's no chain, we just use the signer as the chain, which was the previous behavior.